### DISCUSSION TITLE
TypeError: Cannot read properties of undefined (reading `'jarPath')"

### DISCUSSION THREAD

**Discussion by pronithg** [eclipse-emfcloud/emfcloud#220] [ModelServer]
Hello,

We added "@eclipse-emfcloud/modelserver-theia": "^0.8.0-theia-cr03",   "@eclipse-glsp/theia-integration": "^1.0.0-theia1.34.0" in dependencies on base Theia project after start the error eccored `"ERROR` Could not initialize contribution TypeError: Cannot read properties of undefined (reading `'jarPath')"`

what is the problem we want to add Ecore editor in base Theia
Thank you

**Comment by eneufeld** [comment]
Hi @pronithg ,

I have to admin I'm a bit lost of what you want to do.
Do you want to add this Ecore Editor: https://github.com/eclipse-emfcloud/ecore-glsp/ ?
First of all, this repo is a demonstrator of what is possible. The code there is unfortunately a bit outdated.
Second, if you check the browser-app there you need at least the 4 packages mentioned here: https://github.com/eclipse-emfcloud/ecore-glsp/blob/master/client/browser-app/package.json#L6-L9C46

Best,
Eugen

**Comment by pronithg** [comment]
Hi,
Firstly, I want to integrate emf Ecore editor to base theai. Is there any example package.json file for this.
second we want to integrate prepared emf model to Theia as a server.

By the way 4 packages you mentioned cannot be  installed. message is below
....% >yarn
yarn install v1.22.19
info No lockfile found.
[1/4] 🔍  Resolving packages...
error An unexpected error occurred: "https://registry.yarnpkg.com/@eclipse-emfcloud%2fecore-glsp-common: Not found".

**Comment by eneufeld** [comment]
Hi @pronithg, 

sorry for the late reply.

Yes the packages are not published on npm. If you want to consume ecore-glsp in your own application you need to copy the packages.

So your goal was/is to edit the emf model in theia?

Best,
Eugen
